ODBP.Server/Features/Sitemap/SitemapInstances/SitemapController.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.Mvc;
+using ODBP.Apis.Odrc;
+
+namespace ODBP.Features.Sitemap.SitemapInstances
+{
+    [ApiController]
+    public class SitemapController(IOdrcClientFactory clientFactory, BaseUri baseUri)
+    {
+        const string ApiVersion = "v1";
+        const string ApiRoot = $"/api/{ApiVersion}";
+        const string OrganisatiesPath = $"{ApiRoot}/organisaties?isActief=alle";
+        const string InformatieCategorieenPath = $"{ApiRoot}/informatiecategorieen";
+        const string PublicatiesPath = $"{ApiRoot}/publicaties?publicatiestatus=gepubliceerd&sorteer=registratiedatum";
+        const string DocumentenRoot = $"{ApiRoot}/documenten";
+        const string DocumentenQueryPath = $"{DocumentenRoot}?publicatiestatus=gepubliceerd&sorteer=creatiedatum";
+
+        [HttpGet(ApiRoutes.Sitemap)]
+        public async Task<IActionResult> Get(CancellationToken token)
+        {
+            using var client = clientFactory.Create("sitemap opbouwen");
+            var organisatiesTask = GetWaardelijst(client, OrganisatiesPath, token);
+            var informatiecategorieenTask = GetWaardelijst(client, InformatieCategorieenPath, token);
+
+            var publicaties = await GetPublicaties(client, token);
+            var organisaties = await organisatiesTask;
+            var informatiecategorieen = await informatiecategorieenTask;
+
+            var urls = new List<Publicatie>();
+
+            await foreach (var item in GetAllPages(client, DocumentenQueryPath, token))
+            {
+                var document = item.Deserialize(SitemapPublicatieContext.Default.OdrcDocument);
+                if (document == null ||
+                    !publicaties.TryGetValue(document.Publicatie, out var publicatie) ||
+                    !organisaties.TryGetValue(publicatie.Publisher, out var publisher))
+                {
+                    continue;
+                }
+                urls.Add(new()
+                {
+                    Loc = new Uri(baseUri, $"{DocumentenRoot}/{document.Uuid}/download").ToString(),
+                    Lastmod = Max(document.LaatstGewijzigdDatum, publicatie.LaatstGewijzigdDatum).ToString("o"),
+                    Document = new()
+                    {
+                        DiWoo = new()
+                        {
+                            Publisher = publisher,
+                            Titelcollectie = new()
+                            {
+                                OfficieleTitel = document.OfficieleTitel,
+                            },
+                            Classificatiecollectie = new()
+                            {
+                                Informatiecategorieen = Lookup(publicatie.InformatieCategorieen, informatiecategorieen).ToArray()
```

Comment:
nog zo'n een voorbeeld. we hebben een lijst opgehaald van alle infocategorieren, om de infocategorien die aan deze publicatie hangen maar niet in de officiele lijst zitten eruit te filteren ofzo? of is dit omdat de informatiecategoriegegevens die aan de publicatie hangen niet de vorm of inhoud hebben die in de sitemap moet komen en dit wordt gebruikt om de juiste gegevens erbij te halen? ik zit maar wat te raden. Kan je duidelijker maken wat er allemaal gebeurt, met specifiekere functie/variabale namen en comments?